Exiled.Events/Patches/Fixes/KillPlayer.cs

```diff
@@ -0,0 +1,62 @@
+// -----------------------------------------------------------------------
+// <copyright file="KillPlayer.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Server
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    using Exiled.API.Features;
+    using Exiled.Events.EventArgs;
+    using Exiled.Events.Extensions;
+
+    using HarmonyLib;
+
+    using Mirror;
+
+    using PlayerStatsSystem;
+
+    using static Exiled.Events.Events;
+#pragma warning disable SA1313 // Parameter names should begin with lower-case letter
+
+    /// <summary>
+    /// Prefix of KillPlayer action.
+    /// </summary>
+    [HarmonyPatch(typeof(PlayerStats), nameof(PlayerStats.KillPlayer))]
+    internal class KillPlayer
+    {
+        /// <summary>
+        /// Handles processing of kill player to cast to correct damage handler.
+        /// </summary>
+        /// <param name="__instance"> PlayerStat instance. </param>
+        /// <param name="handler"> DamageHandlerBase instance. </param>
+        /// <returns> Always returns true as we want to run original function with correct DamageHandler. </returns>
+        [HarmonyPrefix]
+        public static bool KillPlayerPrefix(PlayerStats __instance, ref DamageHandlerBase handler)
```

Comment:
Can be made private to remove the need for doc, named Prefix to remove attribute, and can be made return type void since it always returns true to run the method

Reply:
@BuildBoy12 Done